### ISSUE TITLE
Failregex: [DPROTO]: traffic temporary blocked

### ISSUE THREAD

**Issue by robijas** [fail2ban/fail2ban#2946]
Hello,

can anyone help me to get proper failregex for this log:

`L 02/21/2021 - 19:13:38: [DPROTO]: traffic temporary blocked from 74.74.74.73 for flooding; PPS=46.014434; WarnLevel=2.025611
`

this datepatern works:

`datepattern = ^L %%m/%%d/%%Y - %%H:%%M:%%S`

ty.

**Comment by sebres** [comment]
For given datepattern it is:
`^: \[DPROTO\]: traffic temporary blocked from <ADDR>`

You can also extend it to capture double point and space, so it would result to:
```ini
datepattern = ^L %%m/%%d/%%Y - %%H:%%M:%%S:\s+
failregex = ^\[DPROTO\]: traffic temporary blocked from <ADDR>
```

**Comment by robijas** [comment]
hello all good but i have one problem , i have enabled email notification on this jail but when mail arrives there is no lines from logs, just ip info, any solution for this?

mail content:
`Hi,

The IP 194.62.29.43 has just been banned by Fail2Ban after
2 attempts against counter-strike-dproto.


Here is more information about 194.62.29.43 :

% This is the RIPE Database query service.
% The objects are in RPSL format.
%
% The RIPE Database is subject to Terms and Conditions.
% See http://www.ripe.net/db/support/db-terms-conditions.pdf

% Note: this output has been filtered.
%       To receive output for a database update, use the "-B" flag.

% Information related to '194.62.29.0 - 194.62.29.127'

% Abuse contact for '194.62.29.0 - 194.62.29.127' is 'abuse@zap-hosting.com'

inetnum:        194.62.29.0 - 194.62.29.127
netname:        DE-zaphosting
descr:          ZAP-Hosting.com - IF YOU WANT MORE POWER
country:        DE
org:            ORG-ZGCK3-RIPE
admin-c:        MK10417-RIPE
tech-c:         MK10417-RIPE
mnt-routes:     COMBAHTON
status:         ASSIGNED PA
mnt-by:         mnt-nl-skylink2-1
created:        2020-05-07T10:07:01Z
last-modified:  2020-05-29T09:23:34Z
source:         RIPE

organisation:   ORG-ZGCK3-RIPE
org-name:       ZAP-Hosting GmbH & Co. KG
org-type:       OTHER
address:        Krokusweg 8a
address:        DE-48165 Muenster
abuse-c:        ZAPC
mnt-ref:        mnt-de-marcel-1
mnt-ref:        mnt-nl-skylink2-1
mnt-by:         mnt-de-marcel-1
created:        2020-04-22T10:53:39Z
last-modified:  2020-05-07T11:11:26Z
source:         RIPE # Filtered
mnt-ref:        mnt-de-skylink-1
mnt-ref:        SKYLINK

person:         Marvin Kluck
address:        ZAP-Hosting GmbH & Co. KG
address:        Krokusweg 9a
address:        DE-48165 Münster
phone:          +49 251 149 811 80
mnt-by:         ACTIVE-MNT
nic-hdl:        MK10417-RIPE
created:        2011-01-26T20:37:48Z
last-modified:  2017-04-28T16:27:02Z
source:         RIPE # Filtered

% Information related to '194.62.29.0/24AS30823'

route:          194.62.29.0/24
origin:         AS30823
mnt-by:         mnt-nl-skylink2-1
created:        2020-05-29T09:54:02Z
last-modified:  2020-05-29T09:54:02Z
source:         RIPE

% This query was served by the RIPE Database Query Service version 1.99 (WAGYU)


Lines containing failures of 194.62.29.43


Regards,

Fail2Ban
`

**Comment by sebres** [comment]
This is action related stuff, you did not provide which action is used to send a mail.
I guess it is `mail-whois-lines` or `sendmail-whois-lines`, and this action uses grep to capture it from the log files.
You can check `fail2ban-client -d | grep counter-strike-dproto` and then take a look at mailing action (actionban command) and try this command line (after replacing the `<ip>` and other tags) in shell.
Anyway this actions are obsolete and you could use `*-matches` (e. g. `sendmail-whois-matches` instead of `sendmail-whois-lines`), which will output the real matched lines, that failures fail2ban has detected as failures (tag `<matches>`) without to grep it from logfile again.

**Comment by robijas** [comment]
`The IP <ip> has just been banned by Fail2Ban after\n<failures> attempts against counter-strike-dproto.\\n\\n\nHere is more information about <ip> :\\n\n`/usr/bin/whois <ip> || echo missing whois program`\\n\\n\nLines containing failures of <ip>\\n";\nlogpath="/dev/null"; grep -m 1000 -E "(^|[^0-9a-fA-F:])$(echo \'<ip>\' | sed \'s/\\./\\\\./g\')([^0-9a-fA-F:]|$)" $logpath | tail -n 1000;\nprintf %b "\\n\nRegards,\\n\nFail2Ban" )`


i think all good there , cuz on other jails mail is working good, with lines . maybe this failregex making problems cuz containig []?

**Comment by sebres** [comment]
> ... logpath="/dev/null" ...

For some reason the logpath is not substituted correctly, so it retains the default value specified in action.
Normally, action_mwl would substitute to following value https://github.com/fail2ban/fail2ban/blob/753fff9c15387d62e32bc9794cb7a4714b9dd359/config/jail.conf#L220-L221
So logpath of action must be overwritten with logpath of jail. 
I don't know why it does not work in your case. Firstly check your config files.
But as already said, this actions are obsolete and must be replaced with `*-matches`.

>  maybe this failregex making problems cuz containig []?

No the action greps for the IP, the regex does not matter.

---

Just by the way - remove the part `.*$` from failregex - this is neither an anchor, nor expected to match at all (basically `.*$` is equivalent to empty regex by search for match in fail2ban):
```diff
- failregex = ^: \[DPROTO\]: traffic temporary blocked from <HOST> for flooding; .*$
+ failregex = ^: \[DPROTO\]: traffic temporary blocked from <HOST> for flooding\b
```

**Comment by robijas** [comment]
my current action setings:

```
action_mwl = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
             %(mta)s-whois-lines[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", logpath=%(logpath)s, chain="%(chain)s"]
```

updated:

```
action_mwl = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
             %(mta)s-whois-matches[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", logpath=%(logpath)s, chain="%(chain)s"]
```


i will update any changes, ty

**Comment by robijas** [comment]
for some reason *-lines not working, but *-matches does:


```
Hi,

The IP 23.185.0.1 has just been banned by Fail2Ban after
1 attempts against counter-strike-dproto.


Here is more information about 23.185.0.1 :


#
# ARIN WHOIS data and services are subject to the Terms of Use
# available at: https://www.arin.net/resources/registry/whois/tou/
#
# If you see inaccuracies in the results, please report at
# https://www.arin.net/resources/registry/whois/inaccuracy_reporting/
#
# Copyright 1997-2021, American Registry for Internet Numbers, Ltd.
#


NetRange:       23.185.0.0 - 23.185.0.255
CIDR:           23.185.0.0/24
NetName:        PANTHEON-IP4
NetHandle:      NET-23-185-0-0-1
Parent:         NET23 (NET-23-0-0-0-0)
NetType:        Direct Assignment
OriginAS:       AS54113
Organization:   Pantheon (PS-747)
RegDate:        2016-11-21
Updated:        2016-11-22
Comment:        https://pantheon.io/
Ref:            https://rdap.arin.net/registry/ip/23.185.0.0


OrgName:        Pantheon
OrgId:          PS-747
Address:        717 California St Fl 3
City:           San Francisco
StateProv:      CA
PostalCode:     94108
Country:        US
RegDate:        2016-07-21
Updated:        2018-03-15
Ref:            https://rdap.arin.net/registry/entity/PS-747


OrgTechHandle: DTS41-ARIN
OrgTechName:   Strauss, David Timothy
OrgTechPhone:  +1-512-577-5827 
OrgTechEmail:  david@davidstrauss.net
OrgTechRef:    https://rdap.arin.net/registry/entity/DTS41-ARIN

OrgTechHandle: PANTH3-ARIN
OrgTechName:   Pantheon Abuse
OrgTechPhone:  +1-415-780-9765 
OrgTechEmail:  abuse@pantheon.io
OrgTechRef:    https://rdap.arin.net/registry/entity/PANTH3-ARIN

OrgNOCHandle: DTS41-ARIN
OrgNOCName:   Strauss, David Timothy
OrgNOCPhone:  +1-512-577-5827 
OrgNOCEmail:  david@davidstrauss.net
OrgNOCRef:    https://rdap.arin.net/registry/entity/DTS41-ARIN

OrgAbuseHandle: PANTH3-ARIN
OrgAbuseName:   Pantheon Abuse
OrgAbusePhone:  +1-415-780-9765 
OrgAbuseEmail:  abuse@pantheon.io
OrgAbuseRef:    https://rdap.arin.net/registry/entity/PANTH3-ARIN

OrgNOCHandle: PANTH3-ARIN
OrgNOCName:   Pantheon Abuse
OrgNOCPhone:  +1-415-780-9765 
OrgNOCEmail:  abuse@pantheon.io
OrgNOCRef:    https://rdap.arin.net/registry/entity/PANTH3-ARIN


#
# ARIN WHOIS data and services are subject to the Terms of Use
# available at: https://www.arin.net/resources/registry/whois/tou/
#
# If you see inaccuracies in the results, please report at
# https://www.arin.net/resources/registry/whois/inaccuracy_reporting/
#
# Copyright 1997-2021, American Registry for Internet Numbers, Ltd.
#


Matches:

L 02/23/2021 - 14:23:41: [DPROTO]: traffic temporary blocked from 23.185.0.1 for flooding; PPS=47.455747; WarnLevel=2.011179


Regards,

Fail2Ban
```


ty, sebres, great support.

**Comment by sebres** [comment]
> my current action setings:
> `%(mta)s-whois-lines[..., logpath=%(logpath)s, ...]`

take a look in my previous comment for example from jail.conf - logpath substitution is enclosed in quotes (`logpath="%(logpath)s"`), because logpath contains brackets, this (as well as some following parameters) are truncated, so whole interpolation gets confused.

> for some reason *-lines not working, but *-matches does:

as already written: this action "will output the real matched lines, that failures fail2ban has detected as failures (tag `<matches>`) without to grep it from logfile again.". And it also does not need logpath parameter at all.

**Comment by robijas** [comment]
i  understand now, but problem is i didnt touch any off those setting in action it was default after instalation.

**Comment by sebres** [comment]
> i didnt touch any off those setting in action

not action, but jail.conf... which version? 
this quotes are added in e651bc7866f7c0cc32db1fe01d7c93abebd303c5 2 years ago (released in >= 0.10.5 ... 0.11.2)

**Comment by robijas** [comment]
v 0.10.2

updated, now all good :)
